script.js

```diff
@@ -38,30 +47,122 @@ function shuffle(array) {
 
 let shuffledColors = shuffle(COLORS);
 
-// this function loops over the array of colors
-// it creates a new div and gives it a class with the value of the color
-// it also adds an event listener for a click for each card
 function createDivsForColors(colorArray) {
   for (let color of colorArray) {
-    // create a new div
     const newDiv = document.createElement("div");
 
-    // give it a class attribute for the value we are looping over
     newDiv.classList.add(color);
-
-    // call a function handleCardClick when a div is clicked on
     newDiv.addEventListener("click", handleCardClick);
 
-    // append the div to the element with an id of game
     gameContainer.append(newDiv);
   }
 }
 
-// TODO: Implement this function!
+function resetOldGame() {
+  setScore(0);
+  document.querySelector("#current-score").innerText = "";
+  document.querySelector("#final-score").innerText = "";
+  document.querySelector("#game-over").innerText = "";
+  document.querySelector("#play-again").innerText = "";
+}
+
+function initializeNewGame() {
+  let game = document.querySelector("#game");
+  game.innerHTML = "";
+  let newDeck = shuffle(COLORS);
+  createDivsForColors(newDeck);
+}
+
+function startGame() {
+  resetOldGame();
+  initializeNewGame();
+}
+
+const chooseCards = function(chosenCard) {
+  if (firstCard === null) {
+    firstCard = chosenCard;
+    setScore(currentScore + 1);
+  } else if (secondCard === null) {
```

Comment:
what happens if none of these if conditions are satisfied? is this a reasonable scenario in the game?

Reply:
Good question. When I implemented this and tested it, There were only three things a user could do:

1. Click on the 'first card'
2. Click on the 'second card'
3. Repeat until the end of the game
4. Click on the 'New Game' button to either start a new game or reset.

Currently, the 'New Game' button is clearing the current score, but not clearing the board and reshuffling. I need to determine where I went wrong. 

In terms of error handling, the user shouldn't be able to do anything else, but it might be a good idea to catch any errors that might occur. Maybe we could discuss this when when we meet.

Reply:
Agreed to throw an Error if both are set alreadt